terminal/src/main/java/org/jline/terminal/TerminalBuilder.java

```diff
@@ -486,4 +491,9 @@ private static Boolean getBoolean(String name, Boolean def) {
     private <S> S load(Class<S> clazz) {
         return ServiceLoader.load(clazz, clazz.getClassLoader()).iterator().next();
     }
+
+    private static final AtomicReference<Terminal> terminalHolder = new AtomicReference<Terminal>(null);
+    public static final void setTerminal(final Terminal terminal) {
```

Comment:
Would you mind moving the `terminalHolder` field below the `SYSTEM_TERMINAL`, rename it to `TERMINAL_HOLDER` as it's a final static field and add a bit of javadoc to the `setTerminal` method in order to explain the result that it shortcut all other declarations on the builder when using `build()`.

Reply:
Done. I also renamed the method `setGlobalTerminal` and added an `@Deprecated` annotation to communicate that this not an api that should be called unless you are really sure you have a good reason to.

Reply:
Aah, shoot. I have to reopen the PR because I deleted my fork so that it would point to https://github.com/sbt/jline3.

Reply:
#558